source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -44,16 +44,29 @@ Http::FilterHeadersStatus Http1BridgeFilter::encodeHeaders(Http::ResponseHeaderM
     chargeStat(headers);
   }
 
-  if (!do_bridging_ || end_stream) {
+  if (!do_bridging_) {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  response_headers_ = &headers;
+  if (end_stream) {
+    // We still need to check if would-be trailers are present as headers
+    // for a gRPC Trailers-Only response. We do not need to set grpc status
```

Comment:
suggest inserting:
` We still need to check if would-be trailers are present as headers for a gRPC Trailers-Only response. Non-zero grpc-status will change the response code.`